#! /bin/sh /usr/share/dpatch/dpatch-run
## 010_debianise_dj_make_chroot by Thijs Kinkhorst <thijs@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Update script to replace choices with Debian values

@DPATCH@
diff -urNad domjudge-3.0.0~/etc/judgehost-config.php domjudge-3.0.0/etc/judgehost-config.php
--- domjudge-3.0.0~/etc/judgehost-config.php	2009-11-10 18:38:59.000000000 +0100
+++ domjudge-3.0.0/etc/judgehost-config.php	2009-11-15 15:18:37.000000000 +0100
@@ -13,7 +13,7 @@
 // leave empty to disable. This example script can be used to support
 // Sun Java with a chroot (edit the script first!).
 // define('CHROOT_SCRIPT', 'chroot-startstop.sh');
-define('CHROOT_SCRIPT', '');
+define('CHROOT_SCRIPT', 'chroot-startstop.sh');
 
 // Maximum seconds available for compiling
 define('COMPILETIME', 30);
diff -urNad domjudge-3.0.0~/judge/chroot-startstop.sh domjudge-3.0.0/judge/chroot-startstop.sh
--- domjudge-3.0.0~/judge/chroot-startstop.sh	2009-07-12 19:27:50.000000000 +0200
+++ domjudge-3.0.0/judge/chroot-startstop.sh	2009-11-15 15:18:37.000000000 +0100
@@ -19,7 +19,7 @@
 SUBDIRMOUNTS="etc lib usr"
 
 # Location where to bind mount from:
-CHROOTORIGINAL="/chroot/domjudge"
+CHROOTORIGINAL="/var/lib/domjudge/javachroot"
 
 case "$1" in
 	start)
diff -urNad domjudge-3.0.0~/misc-tools/dj_make_chroot domjudge-3.0.0/misc-tools/dj_make_chroot
--- domjudge-3.0.0~/misc-tools/dj_make_chroot	2009-11-15 15:12:43.000000000 +0100
+++ domjudge-3.0.0/misc-tools/dj_make_chroot	2009-11-15 15:19:43.000000000 +0100
@@ -16,8 +16,8 @@
 set -e
 
 # Read command-line parameters:
-CHROOTDIR=$1
-ARCH=$2
+CHROOTDIR=/var/lib/domjudge/javachroot.new
+ARCH=`dpkg --print-architecture`
 
 # List of possible architectures to install chroot for:
 ARCHLIST="alpha,amd64,arm,armel,hppa,i386,ia64,mips,mipsel,powerpc,s390,sparc"
@@ -65,6 +65,7 @@
 [ -z "$ARCH" ]      && error "No architecture given."
 [ -e "$CHROOTDIR" ] && error "'$CHROOTDIR' already exists, remove manually."
 
+rm -rf "$CHROOTDIR"
 mkdir -p "$CHROOTDIR"
 cd "$CHROOTDIR"
 CHROOTDIR="$PWD"
@@ -174,4 +175,6 @@
 
 umount "$CHROOTDIR/proc"
 
+mv /var/lib/domjudge/javachroot.new /var/lib/domjudge/javachroot
+
 exit 0
